gridstatus/spp.py

```diff
@@ -355,6 +352,119 @@ def get_load_forecast_mid_term(
+            tuple: (floored_date, is_repeated_hour)
+        """
+        is_repeated_hour = False
+        try:
+            floored_date = date.floor(freq)
+        except pytz.AmbiguousTimeError:
+            is_repeated_hour = True
+            floored_date = self.safe_for_dst_transition_floor(date, freq)
+
+        return floored_date, is_repeated_hour
+
+    def _get_short_term_forecast_data(
+        self,
+        date: str | pd.Timestamp,
+        base_url: str,
+        file_prefix: str,
+        buffer_minutes: int = 2,
+    ) -> tuple[pd.DataFrame, str] | None:
+        """Get short-term forecast data with common DST handling logic.
+
+        Args:
+            date: Date to get data for. Supports "latest" and "today"
+            base_url: Base URL for downloads
+            file_prefix: Prefix for the file name (e.g., "OP-STLF", "OP-STRF")
+            buffer_minutes: Buffer minutes for "latest" date
+
+        Returns:
+            tuple: (dataframe, url) or None if date is in the future
+        """
+        if date == "latest":
+            date = self.now() - pd.Timedelta(minutes=buffer_minutes)
+
+        # Files do not exist in the future
+        if date > self.now():
+            return None
+
+        floored_date, is_repeated_hour = self._handle_dst_floor_date(date, "5min")
+
+        hour = floored_date.hour
+        padded_hour = str(hour).zfill(2)
+        padded_hour_plus_one = str((hour + 1) % 24).zfill(2)
+
+        # The 100 file does not have a "d" but 105d through 155d do.
+        add_d = is_repeated_hour and not floored_date.minute == 0
+
+        # The first hour in the URL is 1 after the hour in the filename.
+        url = base_url + floored_date.strftime(
+            f"/%Y/%m/%d/{padded_hour_plus_one}/{file_prefix}-%Y%m%d{padded_hour}%M{'d' if add_d else ''}.csv",
+        )
+
+        logger.info(f"Downloading {url}")
+        df = pd.read_csv(url)
+
+        return df, url
+
+    def _get_mid_term_forecast_data(
+        self,
+        date: str | pd.Timestamp,
+        base_url: str,
+        file_prefix: str,
+        buffer_minutes: int = 10,
+    ) -> tuple[pd.DataFrame, str] | None:
+        """Get mid-term forecast data with common DST handling logic.
+
+        Args:
+            date: Date to get data for. Supports "latest" and "today"
+            base_url: Base URL for downloads
+            file_prefix: Prefix for the file name (e.g., "OP-MTLF", "OP-MTRF")
+            buffer_minutes: Buffer minutes for "latest" date
+
+        Returns:
+            tuple: (dataframe, url) or None if date is in the future
+        """
+        if date == "latest":
+            date = self.now() - pd.Timedelta(minutes=buffer_minutes)
+
+        # Files do not exist in the future
+        if date > self.now():
+            return None
+
+        floored_date, _is_repeated_hour = self._handle_dst_floor_date(date, "h")
+
+        # For mid-term hourly forecasts during 2025 DST end there is a 0200d file.
```

Comment:
Only the 0200d file has a "d". Note there is a missing hour because there is not also a 0200 file.

<img width="2136" height="882" alt="Image" src="https://github.com/user-attachments/assets/31f4e420-db4d-413d-b171-f24dbab97a01" />
<img width="2112" height="546" alt="Image" src="https://github.com/user-attachments/assets/82da78f7-67df-46a6-8932-4d59fbc8b832" />